src/main/java/com/matera/cadastrocentral/address/AddressService.java

```diff
@@ -0,0 +1,45 @@
+package com.matera.cadastrocentral.address;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class AddressService {
+
+    @Autowired
+    private AddressRepository addressRepository;
+
+    public Address getAddressById(UUID id) {
+        return addressRepository.findById(id).orElse(null);
+    }
+
+    public List<Address> getAddressListByClientId(UUID clientId) {
+        return addressRepository.findAllByClientId(clientId);
+    }
+
+    public Address insertAddress(Address newAddress) {
+        return addressRepository.save(newAddress);
+    }
+
+    public void deleteAddress(UUID id) {
+        addressRepository.deleteById(id);
+    }
+
+    public Address alterAddress(UUID id, Address newAddress) {
+        Address alteredAddress = addressRepository.findById(id).orElse(null);
+
+        alteredAddress.setReference(Optional.ofNullable(newAddress.getReference()).orElse(alteredAddress.getReference()));
```

Comment:
if alteredAddress is null we will get a NPE(NullPointerException) here, maybe check if its null then return null before reaching this code is a good idea